stan/math/prim/meta/broadcast_array.hpp

```diff
@@ -30,22 +32,34 @@ class broadcast_array {
   void operator=(const Y& m) {
     prim_ = m[0];
   }
+  explicit broadcast_array(const broadcast_array<T>& other) = default;
```

Comment:
If these are default, why do they need defined?

Reply:
For some reason in the  `reduce_sum` PR we did
```
broadcast_array& operator=(const broadcast_array& other) = default;
```

I'm not sure why and can delete these or (@wds15 did you put the `operator=` one here for a reason?

Reply:
Maybe. Likely not needed any longer.